### DISCUSSION TITLE
Potential conflict with AUTOMATIC1111's Stable Diffusion webui (Windows 10)

### DISCUSSION THREAD

**Discussion by BangDroid** [hyperion-project/hyperion.ng#1785] [General]
I understand I may be in a very small minority edge case of people who run A1's SD webui on the same machine as Hyperion, but I am wondering if anyone else has encountered problems running both on the same machine. Or even running other python projects on the same machine as the one running Hyperion.

I don't feel I have enough information to open an Issue yet. The reason I suspect a conflict between Hyperion and A1's Stable Diffusion webui is mainly the python commonality. Hyperion has been extremely reliable and has run for hundred maybe thousands of hours without issue, I barely even think about it, it just works. But recently I have been dabbling with Stable Diffusion (which is python and uses pyenv) and occasionally after X amount of time (X being seemingly random), Hyperion will completely freeze and become unresponsive. Lights go out, the settings web server will not load and times out in the browser. Even after ending hyperiond.exe in task manager and relaunching, it remains frozen and unresponsive. The only way to recover from this is a complete system reboot. Which of course means there's nothing in the log about the last issue when I check.

I haven't had any issues with SD at all, nor noticed anything in any log on that side of things.
I am using Hyperion with _Hyperion Screen Capture_ by _Sabaatworld_ and the only thing I have noticed is in it's log when Hyperion freezes I get
```
2024-08-23 09:34:56 [ERROR] [6] HyperionScreenCap.Helper.HyperionTask.StartCapture:0 - HyperionTask[ConfigurationId: 4792cd]: Exception in screen capture attempt: 1
System.Exception: Error occured while sending image to server: Unable to write data to the transport connection: A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond.
```
Which I think only means that Hyperion Screen Capture is failing to communicate with Hyperion because it's frozen.
Hyperion freezing like this only happens if I launch Stable Diffusion, but not always. Other than that Hyperion is exceptionally reliable.

I understand how extremely edge case this is and why is really doesn't deserve much attention from the devs, but I would like to leave the discussion here for anyone else who may be facing similar issues, or potentially has any insight.

**Comment by Lord-Grey** [comment]
What Python version are you running?
There are known issues that Hyperion does not start with Python 12.
You should start Hyperion with Python 11.

**Reply by BangDroid** [reply]
From command line:
```
python --version
Python 3.10.6
```
But I'm doubting that's the issue because as mentioned, Hyperion has been exceptionally stable prior to this, and continues to be so long as I don't start up SD. Even then it's sporadic. I should also note that I'm running Hyperion 2.0.12.
A1's SD also uses venv and has it's own bundled python versions. I also have other various software that use other versions of python and venv bundled with them, e.g. Thonny, Mu editor, KiCad, GIMP, Blender. None of these have cause issues with Hyperion either.

I have also noticed just now there is _Python35-32, Python39_ and _Python310_ directories in my _AppData\Local\Python_ dir, as well as _Python310_ and _Python39_ both defined in my PATH with 310 being first. This may be problematic?

I have also considered that it's just a consequence of running such a resource intensive task like Stable Diffusion, USB communication might be getting interupted and crashing the Arduino, which may freeze Hyperion. Or a GPU issue related to screen capture while under intense load.

**Comment by Lord-Grey** [comment]
My first thought was also unavailable resources.
Maybe you check the consumption when it reoccurs.

**Comment by BangDroid** [comment]
Hyperion didn't fully freeze this time, but the Arduino disconnected while generating an image with SD. I managed to check Hyperion's log and found the following:
```
2024-08-25T11:07:54.051Z [ LEDDEVICE] (DEBUG) (ProviderRs232.cpp:115:ProviderRs232::close()) Flush was successful
2024-08-25T11:07:54.051Z [ LEDDEVICE] (DEBUG) (ProviderRs232.cpp:117:ProviderRs232::close()) Close UART: COM7
2024-08-25T11:07:54.051Z [ LEDDEVICE] (ERROR) Device disabled, device 'adalight' signals error: 'Rs232 SerialPortError: Access is denied.'
2024-08-25T11:07:54.051Z [ COMPONENTREG] (DEBUG) (ComponentRegister.cpp:41:ComponentRegister::setNewComponentState()) LED device: disabled
```
I'm guessing resource contention causes this while generating. My Arduino Nano was connected via a USB 2.0 PCIE expansion slot which in turn connects via the motherboard chipset. I have since plugged it into a USB 3.0 port with direct CPU connection, so hopefully this resolves the resource contention. I'll see how it goes over the next few days.
